crates/network-scanner/examples/scan.rs

```diff
@@ -33,15 +34,21 @@ fn main() -> anyhow::Result<()> {
         let stream = scanner.start()?;
         let stream_clone = stream.clone();
         let now = std::time::Instant::now();
-        while let Ok(Some(res)) = stream_clone
-            .recv_timeout(Duration::from_secs(120))
+
+        tokio::task::spawn_blocking(move || {
```

Comment:
type anything to stop the scanning

Reply:
I recommend this instead: https://docs.rs/tokio/latest/tokio/signal/fn.ctrl_c.html

Reply:
updated